src/vcpkg/binarycaching.cpp

```diff
@@ -826,63 +863,45 @@ namespace
 
         size_t push_success(const BinaryPackageWriteInfo& request, MessageSink& msg_sink) override
         {
+            PrintingDiagnosticContext pdc{msg_sink};
+            WarningDiagnosticContext wdc{pdc};
             auto& spec = request.spec;
-
             auto nuspec_path = m_buildtrees / spec.name() / spec.triplet().canonical_name() + ".nuspec";
+            auto nuspec_contents = request.nuspec.value_or_exit(VCPKG_LINE_INFO);
             std::error_code ec;
-            m_fs.write_contents(nuspec_path, request.nuspec.value_or_exit(VCPKG_LINE_INFO), ec);
+            m_fs.write_contents(nuspec_path, nuspec_contents, ec);
             if (ec)
             {
-                msg_sink.println(Color::error, msgPackingVendorFailed, msg::vendor = "NuGet");
+                wdc.report_error(format_filesystem_call_error(ec, "write_contents", {nuspec_path, nuspec_contents}));
+                wdc.report(DiagnosticLine{DiagKind::Note, msg::format(msgWhilePackingNuGetPackage)});
                 return 0;
             }
 
-            auto packed_result = m_cmd.pack(msg_sink, nuspec_path, m_buildtrees);
+            auto pack_result = m_cmd.pack(wdc, nuspec_path, m_buildtrees);
             m_fs.remove(nuspec_path, IgnoreErrors{});
-            if (!packed_result)
+            if (!pack_result)
             {
-                msg_sink.println(Color::error, msgPackingVendorFailed, msg::vendor = "NuGet");
                 return 0;
             }
 
             size_t count_stored = 0;
             auto nupkg_path = m_buildtrees / make_feedref(request, m_nuget_prefix).nupkg_filename();
             for (auto&& write_src : m_sources)
             {
-                msg_sink.println(msgUploadingBinariesToVendor,
-                                 msg::spec = request.display_name,
-                                 msg::vendor = "NuGet",
-                                 msg::path = write_src);
-                if (!m_cmd.push(msg_sink, nupkg_path, nuget_sources_arg({&write_src, 1})))
-                {
-                    msg_sink.println(Color::error,
-                                     msg::format(msgPushingVendorFailed, msg::vendor = "NuGet", msg::path = write_src)
-                                         .append_raw('\n')
-                                         .append(msgSeeURL, msg::url = docs::troubleshoot_binary_cache_url));
```

Comment:
Previously this replaced all rich error information from NuGet with the generic `msgPushingVendorFailed`